.github/workflows/artifacts-build.yml

```diff
@@ -195,36 +194,349 @@ jobs:
         echo "https://storage.googleapis.com/icu4x-pr-artifacts/gha/${{ github.sha }}/wasm-demo/index.html"
         echo "::endgroup::"
 
+  bench-perf:
+    # This is too expensive to run on every push, so only run it on main.
+    # Remove this line to debug.
+    if: github.ref == 'refs/heads/main' && github.repository == 'unicode-org/icu4x'
+    concurrency:
+      # Allow one run at a time to include the previous run's results
+      group: bench-perf
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        component:
+          - components/locid
+          - components/plurals
+          - components/datetime
+          - components/collections
+          - utils/fixed_decimal
+
+    # If you are modifying and debugging is required, don't be afraid to get
+    # messy in a personal fork, if no better way to do it.
+    # Example "debugging" workflow: https://github.com/echeran/icu4x/actions/runs/296714990
+
+    steps:
+    - uses: actions/checkout@v3
+
+    # Cargo-make boilerplate
+    - name: Get cargo-make version
+      id: cargo-make-version
+      run: |
+        echo "hash=$(cargo search cargo-make | grep '^cargo-make =' | md5sum)" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Attempt to load cached cargo-make
+      uses: actions/cache@v3
+      id: cargo-make-cache
+      with:
+        path: |
+          ~/.cargo/bin/cargo-make
+          ~/.cargo/bin/cargo-make.exe
+        key: ${{ runner.os }}-make-${{ steps.cargo-make-version.outputs.hash }}
+    - name: Install cargo-make
+      if: steps.cargo-make-cache.outputs.cache-hit != 'true'
+      run: cargo +stable install cargo-make
+
+    # Actual job
+
+    - name: Run benchmark
+      run: |
+        pushd $PWD && cd ${{ matrix.component }};
+        export REL_OUTPUT_PATH="`dirs +1`/benchmarks/perf/${{ matrix.component }}";
+        mkdir -p $REL_OUTPUT_PATH;
+        export OUTPUT_PATH_CMD="ls -d $REL_OUTPUT_PATH";
+        export OUTPUT_PATH=$(echo $OUTPUT_PATH_CMD | sh);
+        cargo bench --features bench -- --output-format bencher | tee $OUTPUT_PATH/output.txt;
+
+    - name:  Download previous benchmark data
```

Comment:
I've changed this to use the GCP bucket for a couple of reasons:
* Using the `unmerged-pr-bench-data` branch for PR-level benches is my suspect for why the `datasize` and `locid` benches are reporting errors. I *think* they are using the previous commit on that branch as the baseline data, but that data could be from another PR, or even from the future invocation on the same PR (there can be all kinds of race conditions).
* Using Git branches to store this data is unnecessary, as we only ever need the latest version (it includes the whole history). Having O(10k) commits on these branches makes cloning the repo expensive, and `git log` hard to handle.
* There's no easy way to associate a data commit on the `merged-bench-data` branch with its commit. `main` *might* correspond to `merged-bench-data`, but it takes ~half an hour for that to update, and there can be race conditions if multiple PRs are merged.